.github/workflows/sync-onnxrt-main.yaml

```diff
@@ -1,46 +1,61 @@
 name: Onnxruntime main weekly sync
 on:
   schedule:
-    - cron: "05 17 * * 1"
+    - cron: '27 1 * * 4'
 
 jobs:
-  runs-on: ubuntu-latest
-  sync:
+  createPullRequest:
+    name: Update and create pull request
+    runs-on: self-hosted
     steps:
-      - uses: actions/checkout@v3
-        with:
-          ref: develop
-          path: ../
+      - name: cleanup-old-runs
+        run: rm -rf onnxruntime
 
-  get_date:
-    steps:
-      - run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+      - name: get_date
+        run: echo todays_date="$(date +'%Y-%m-%d')" >> $GITHUB_ENV
 
-  update_file:
-    needs: [sync get_date]
-    steps:
-      - run: git clone https://github.com/microsoft/onnxruntime.git && cd onnxruntime && git rev-parse HEAD >> ../test/onnx/.onnxrt-commit
+      - name: get-onnxrt-main
+        run: git clone https://github.com/microsoft/onnxruntime.git
+        
+      - name: extract_sha1 
+        run: cd onnxruntime && echo onnxsha="$(git rev-parse HEAD)" >> $GITHUB_ENV
```

Comment:
git ls-remote https://github.com/microsoft/onnxruntime.git HEAD | awk '{ print $1}' >> $GITHUB_ENV

Reply:
This is a great suggestion. I didn't realize we could do that.